SwiftCompilerSources/Sources/Optimizer/FunctionPasses/CopyToBorrowOptimization.swift

```diff
@@ -124,8 +124,11 @@ private struct Uses {
   // E.g. the none-case of a switch_enum of an Optional.
   private(set) var nonDestroyingLiverangeExits: Stack<Instruction>
 
-  var allLifetimeEndingInstructions: [Instruction] {
-    Array(destroys.lazy.map { $0 }) + Array(nonDestroyingLiverangeExits)
+  // FIXME: Use an InstructionSetVector
+  var allLifetimeEndingInstructions : InstructionWorklist {
+    var worklist = InstructionWorklist(context)
+    worklist.pushIfNotVisited(contentsOf: Array(destroys.lazy.map { $0 }) + Array(nonDestroyingLiverangeExits))
```

Comment:
You don't need to "convert" the lazy map sequence to an array, because `pushIfNotVisited` accepts any kind of sequence.
Instead you can call `pushIfNotVisited` twice:
```
  worklist.pushIfNotVisited(contentsOf: destroys.lazy.map { $0 })
  worklist.pushIfNotVisited(contentsOf: nonDestroyingLiverangeExits)
```